File: app/Models/Tenant/Flat.php
<?php

namespace App\Models\Tenant;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Flat extends Model
{
    use SoftDeletes;

    protected $connection = 'society';

    protected $fillable = [
        'block_id',
        'flat_number',
        'floor_number',
        'flat_type',
        'area_sqft',
        'ownership_type',
        'owner_name',
        'status',
        'car_parking_slot',
        'bike_parking_slot',
        'amenities',
        'metadata',
    ];

    protected $casts = [
        'amenities' => 'array',
        'metadata' => 'array',
    ];

    public function block(): BelongsTo
    {
        return $this->belongsTo(Block::class);
    }

    public function residents(): HasMany
    {
        return $this->hasMany(FlatResident::class);
    }

    public function maintenanceRequests(): HasMany
    {
        return $this->hasMany(MaintenanceRequest::class);
    }

    public function scopeActive($query)
    {
        return $query->where('status', 'active');
    }

    public function getDisplayLabelAttribute(): string
    {
        return $this->block ? "{$this->block->name} / {$this->flat_number}" : $this->flat_number;
    }
}

File: mobile/lib/core/models/flat.dart
class Flat {
  Flat({
    required this.id,
    required this.flatNumber,
    this.floorNumber,
    this.flatType,
    this.areaSqft,
    this.ownerName,
    required this.displayLabel,
    this.blockId,
    this.blockName,
  });

  factory Flat.fromJson(Map<String, dynamic> json) {
    final block = json['block'] as Map<String, dynamic>?;

    return Flat(
      id: json['id'] as int,
      flatNumber: json['flat_number'] as String,
      floorNumber: json['floor_number']?.toString(),
      flatType: json['flat_type'] as String?,
      areaSqft: (json['area_sqft'] as num?)?.toDouble(),
      ownerName: json['owner_name'] as String?,
      displayLabel: json['display_label'] as String? ?? json['flat_number'] as String,
      blockId: block?['id'] as int?,
      blockName: block?['name'] as String?,
    );
  }

  final int id;
  final String flatNumber;
  final String? floorNumber;
  final String? flatType;
  final double? areaSqft;
  final String? ownerName;
  final String displayLabel;
  final int? blockId;
  final String? blockName;
}
